Userland/Utilities/unzip.cpp

```diff
@@ -19,17 +19,21 @@
 
 static bool unpack_zip_member(Archive::ZipMember zip_member, bool quiet)
 {
+    char null_terminated_name[zip_member.name.bytes_as_string_view().length() + 1];
+    memcpy(null_terminated_name, zip_member.name.bytes_as_string_view().characters_without_null_termination(), zip_member.name.bytes_as_string_view().length());
+    null_terminated_name[zip_member.name.bytes_as_string_view().length()] = 0;
+
     if (zip_member.is_directory) {
-        if (mkdir(zip_member.name.characters(), 0755) < 0) {
+        if (mkdir(null_terminated_name, 0755) < 0) {
             perror("mkdir");
             return false;
         }
         if (!quiet)
             outln(" extracting: {}", zip_member.name);
         return true;
     }
-    MUST(Core::Directory::create(LexicalPath(zip_member.name).parent(), Core::Directory::CreateDirectories::Yes));
-    auto new_file = Core::File::construct(zip_member.name);
+    MUST(Core::Directory::create(LexicalPath(null_terminated_name).parent(), Core::Directory::CreateDirectories::Yes));
```

Comment:
Might be worth waiting on #16371 to avoid memcpy-ing the entire string just to null-terminate it and shove it back into a `DeprecatedString` for `LexicalPath` to use.